superset/models/alerts.py

```diff
@@ -100,3 +119,187 @@ class AlertLog(Model):
     @property
     def duration(self) -> int:
         return (self.dttm_end - self.dttm_start).total_seconds()
+
+
+class Observer(Model):
+
+    __tablename__ = "alert_observers"
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String(150), nullable=False)
+    observer_type = Column(Enum(AlertObserverType))
+    validation_type = Column(Enum(AlertValidationType))
+
+    @declared_attr
+    def alert_id(self) -> int:
+        return Column(Integer, ForeignKey("alerts.id"), nullable=False)
+
+    @declared_attr
+    def alert(self) -> RelationshipProperty:
+        return relationship(
+            "Alert",
+            foreign_keys=[self.alert_id],
+            backref=backref("alert_observers", cascade="all, delete-orphan"),
+        )
+
+    @declared_attr
+    def database_id(self) -> int:
+        return Column(Integer, ForeignKey("dbs.id"), nullable=False)
+
+    @declared_attr
+    def database(self) -> RelationshipProperty:
+        return relationship(
+            "Database",
+            foreign_keys=[self.database_id],
+            backref=backref("alert_observers", cascade="all, delete-orphan"),
+        )
+
+    __mapper_args__ = {
+        "polymorphic_identity": "base_observer",
+        "polymorphic_on": observer_type,
+    }
+
+
+class SQLObserver(Observer):
+
+    __tablename__ = "alert_observers"
+
+    sql = Column(Text, nullable=False)
+
+    def query(self) -> None:
+        parsed_query = ParsedQuery(self.sql)
+        sql = parsed_query.stripped()
+        df = self.database.get_df(sql)
+
+        self.observations.append(  # pylint: disable=no-member
+            Observation(dttm_ts=datetime.utcnow(), value=df.to_json())
+        )
+
+        db.session.commit()
+
+    def get_observations(self, observation_num: int) -> List[Any]:
```

Comment:
add a todo to abstract observations form the sqlamodels
e.g. follow the example of the: https://github.com/apache/incubator-superset/blob/671461d0d03ccc7ac1d7bef9cf1a5d9c1f39fdeb/superset/utils/log.py#L32